pkg/registration/product.go

```diff
@@ -2,46 +2,89 @@ package registration
 
 import (
 	"encoding/json"
+	"fmt"
+	"strings"
 
 	"github.com/SUSE/connect-ng/pkg/connection"
 )
 
-// Product as defined from SCC'S API.
+// Product as defined from SCC's API.
 type Product struct {
+	// NOTE: what in SCC's API is called "identifier", it is the "name" in
+	// zypper's nomenclature, but that's different from SCC's "name". Hence,
+	// internally we will be consistent with SCC's API and name this attribute
+	// "identifier", regardless of the naming in zypper.
+	Identifier string `xml:"name,attr" json:"identifier"`
 	Name       string `json:"name"`
-	Identifier string `json:"identifier"`
-	Version    string `json:"version"`
-	Arch       string `json:"arch"`
-	Summary    string `json:"summary,omitempty"`
-	IsBase     bool   `json:"isbase"`
+
+	Version string `xml:"version,attr" json:"version"`
+	Arch    string `xml:"arch,attr" json:"arch"`
+	Release string `xml:"release,attr" json:"-"`
+	Summary string `xml:"summary,attr" json:"summary,omitempty"`
+	IsBase  bool   `xml:"isbase,attr" json:"isbase"`
 
 	FriendlyName string `json:"friendly_name,omitempty"`
-	ReleaseType  string `json:"release_type,omitempty"`
+	ReleaseType  string `xml:"registerrelease,attr" json:"release_type,omitempty"`
 	Available    bool   `json:"available"`
 	Free         bool   `json:"free"`
 	Recommended  bool   `json:"recommended"`
 
-	// optional extension products
+	// Optional extension products
 	Extensions []Product `json:"extensions,omitempty"`
 
-	Description  string       `json:"description,omitempty"`
+	Description  string       `xml:"description" json:"description,omitempty"`
 	EULAURL      string       `json:"eula_url,omitempty"`
 	FormerName   string       `json:"former_identifier,omitempty"`
 	ProductType  string       `json:"product_type,omitempty"`
+	ProductLine  string       `xml:"productline,attr"`
```

Comment:
Do you have a clue why we need this? I have not seen any application of this information but probably I just missed it

Reply:
So, this is part of the Product and ZypperProduct merge. No idea why it's needed, but since it comes from zypper and the old ZypperProduct used it (see further that it comes by parsing XML), then I thought to leave it.

Reply:
fair enough, was just curious if you know why this is there. I don't.. Lets keep it then.